### PR TITLE
[hp] reauthenticate

### PR DESCRIPTION
**Pull request by gogolok**
The current hp (storage) provider doesn't support token expiration.

We have a token expiration time of around ~50 minutes when using our Swift service.
Afterwards the fog gem will throw an exception.

http://docs.openstack.org/api/quick-start/content/
```
2. Send API requests and include the token in the X-Auth-Token header. Continue to send API requests with that token until the job completes or a 401 Unauthorized error occurs.
3. If the 401 Unauthorized error occurs, request another token.
```

This PR fixes the hp (storage) provider.

When creating the request it's important the original `params` won't be modified so that the `retry` will work.

### CODE REVIEW

**Comment by geemus** [comment]
@mwhagedorn thanks, good to know.

File: lib/fog/hp/storage.rb [hunk 1]
```diff
@@ -303,55 +303,37 @@ class Real
...
-          else
-            # Call the legacy v1.0/v1.1 authentication
-            credentials = Fog::HP.authenticate_v1(options, @connection_options)
-            # the user sends in the cdn endpoint
-            @hp_storage_uri = options[:hp_auth_uri]
-            @hp_cdn_uri  = options[:hp_cdn_uri]
-          end
-
-          @auth_token = credentials[:auth_token]
+          authenticate(@options)
```
Comment (plribeiro3000) [inline lib/fog/hp/storage.rb]: Why do you need to pass `@options` here?
I think you can just use `@options` in the context of `authenticate`, right?

**Comment by plribeiro3000** [comment]
@gogolok Could you rebase this PR again please?

**Comment by plribeiro3000** [comment]
Closing due inactivity. Comment here when you have time to work on top of this again.

**Comment by geemus** [comment]
Thanks!

**Pull request closed without merging**